pkg/controller/replication/conversion.go

```diff
@@ -188,6 +190,44 @@ func (c clientsetAdapter) Extensions() extensionsv1beta1client.ExtensionsV1beta1
 	return conversionExtensionsClient{c.Interface, c.Interface.Extensions()}
 }
 
+func (c clientsetAdapter) AppsV1beta2() appsv1beta2.AppsV1beta2Interface {
+	return conversionAppsV1beta2Client{c.Interface, c.Interface.AppsV1beta2()}
+}
+
+func (c clientsetAdapter) AppsV1() appsv1.AppsV1Interface {
+	return conversionAppsV1Client{c.Interface, c.Interface.AppsV1()}
+}
+
+func (c clientsetAdapter) Apps() appsv1.AppsV1Interface {
+	return conversionAppsV1Client{c.Interface, c.Interface.AppsV1()}
+}
+
+type conversionAppsV1beta2Client struct {
+	clientset clientset.Interface
+	appsv1beta2.AppsV1beta2Interface
+}
+
+func (c conversionAppsV1beta2Client) ReplicaSets(namespace string) appsv1beta2.ReplicaSetInterface {
+	// TODO(enisoc): This will force RC integration tests to fail if anyone tries to update
+	// ReplicaSetController to use apps/v1beta2 without updating this conversion adapter.
+	// Please change conversionClient to use the new RS version instead of extensions/v1beta1,
+	// and then return a conversionClient here.
+	panic("need to update RC/RS conversionClient for apps/v1beta2")
+}
+
+type conversionAppsV1Client struct {
+	clientset clientset.Interface
+	appsv1.AppsV1Interface
+}
+
+func (c conversionAppsV1Client) ReplicaSets(namespace string) appsv1.ReplicaSetInterface {
+	// TODO(enisoc): This will force RC integration tests to fail if anyone tries to update
+	// ReplicaSetController to use apps/v1 without updating this conversion adapter.
+	// Please change conversionClient to use the new RS version instead of extensions/v1beta1,
+	// and then return a conversionClient here.
+	panic("need to update RC/RS conversionClient for apps/v1")
```

Comment:
This cannot protect us if a new api version of RS is added (e.g. v2beta1) and RS controller was switched to use that new version. However I don't expect us doing that in the near future, so this is fine.

Reply:
I think the adoption integration test should still fail in that case, since it will look for the RC under RS storage and get NotFound. I mostly added these panics to give a more obvious error message when possible.